### ISSUE TITLE
In-app Notifications for upvotes

### ISSUE THREAD

**Issue by sachben91** [hicommonwealth/commonwealth#8797]
### Description

The new knock notifications do not currently have notification for upvotes. Users should get notifications for upvotes on their threads and comments 

### User Stories / Acceptance Criteria

- User should be subscribed to upvotes on their thread or comment upon creating the thread or comment
-  User should receive an in-app notification if there has been an upvote on their thread or comment that they made on a thread 
- On clicking the notification the user should be taken to the thread or comment that is relevant to the notification 
- User should be unsubscribed from the reactions when they unsubscribe from the thread or comment 
- Notification copy "[username] liked [truncated body of the comment or thread]" 
- copy can be truncated at around 25 characters or whatever looks best in the UI upon testing 

### Additional Context
Knock Workflow https://dashboard.knock.app/common-dev/development/workflows/new-upvote

**Comment by timolegros** [comment]
@sachben91 can you confirm whether users who subscribe to a thread/comment should also receive upvote notifications or are upvote notifications only for the original author (who is subscribed)? To me only sending upvote notifications to the original author makes sense.

**Comment by sachben91** [comment]
@timolegros upvote notifications only for the author makes sense here.

**Comment by timolegros** [comment]
@sachben91 the notification copy does not work well with batching functionality. I could use the last reactions author but I think something like "X recent upvotes on your thread "[Truncated Title]" would be better. 

Additionally, I checked the workflow in Knock and I think we can improve it as follows:
- 3-hour fixed window
- 24-hour max extension (each event received in a batch adds 6 hours to batch time up to max of 24)
- Throttle by the recipient and by subscribed object per day (user gets maximum 1 upvote notification per object per day)
- Throttle by recipient alone (user gets maximum 2 upvote notifications per day)

With this setup, users are limited to 2 upvote notifications per day regardless of the total number of upvotes. Additionally, users will never receive more than 1 notification for upvotes on a single object per day. The sliding window captures as many votes as possible every day. If 3 hours after the initial/first upvote, the user has not received another upvote, the single upvote notification is sent. On the other hand, if a user receives 1 upvote every 3 hours, the user will receive a single notification after 24 hours that includes all the upvotes.

Let's pair up on August 27 to do this and review template editing.

Side Note:
We may also want to add "Send Windows" and in-app and push notification workflow steps. If we set the user's timezone on Knock we can set reasonable times at which notifications should be sent.
